packages/core/src/kernelCoreTemplate.js

```diff
@@ -82,6 +90,44 @@
     Object.freeze(kernel)
     return kernel
 
+    function scuttleGlobalThis(extraAvoids = new Array()) {
+      let props = Object.getOwnPropertyNames(globalThis)
```

Comment:
doesnt track properties of the prototype chain (important?)

Reply:
you mean prototype of `window`? because it does, one line after this line ([95-100](https://github.com/LavaMoat/LavaMoat/pull/360/files#diff-cbec1016c903fe52b8d76860ebf3400e172b83a3374b8796bab149b3721d8064R95-R100))

or did you mean something else?

Reply:
i believe the prototype chain is not consistent across browsers and this also needs to work in node.js

Reply:
in terms of getting a hold on as many props that are accessible via `globalThis` as possible, I couldn't find a better way then this.

I came up with a quick demo to demonstrate that https://github.com/weizman/global-this-props. it also proves to work properly on node as well as all modern browsers.

the reason this is not trivial is because not all props that you can access via `globalThis` are in fact `globalThis`'s own props - there are props that are own props of other objects that are not `globalThis` that do sit on `globalThis`.

For example, `hasOwnProperty` is a property accessible via `globalThis` and can be configured for it, but is not its own prop. `hasOwnProperty` belongs to `Object` and for `Object` it cannot be reconfigured.